hack/run-e2e-conformance-virtual-cluster.sh

```diff
@@ -49,7 +57,7 @@ api_ip: $api_ip
 virtual_router_id: $virtual_router_id
 domain: $domain_name
 ctlplanes: 1
-workers: 2
+workers: $NUM_OF_WORKERS
```

Comment:
To me, this file should be downstream only since it's using kcli to deploy an OCP cluster. That goes for the Makefile targets too. No-one from upstream can use this file without OCP. The only reason to have it here is to reduce the effort of deviating upstream from downstream. However, I think if this file is downstream only, there is nothing to maintain upstream for that.

Reply:
Here my thoughts on this.:

[kcli](https://github.com/karmab/kcli) can be seen as an upstream component, though it is primarily used in some openshift deployment. 

File [`hack/run-e2e-conformance-virtual-cluster.sh`](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/528/files#diff-7e6d81500bdbfcb0341bce1452f665c6c78edc5ab3125d0ac217bde9e7cbe02d) deploys vanilla kubernetes clusters, so it's fine for it to stay u/s.

You're probably referring to [`hack/run-e2e-conformance-virtual-ocp.sh`](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/528/files#diff-90e627cba9db282e04456e7e550bd478d2c12fbbbce4104f9421d640eb3a6447), which deploys and tests PR against an openshift cluster. In this case, having it u/s helps to intercept breaking changes early in the pipeline, as the code has plenty of `IsOpenshiftCluster()`.

Reply:
I understand that we have plenty of `IsOpenshiftCluster()` in upstream, but the question is more fundamental: should we have such things in upstream? It's not black and white either in my opinion. The less of that we have upstream, the better. In this particular case, it seems that we could move that complete file to downstream only.

Reply:
Hi @bn222 take a look we have a CI system running here
https://github.com/k8snetworkplumbingwg/sriov-network-operator/actions/runs/6606102537/job/17942015613?pr=528

https://github.com/k8snetworkplumbingwg/sriov-network-operator/actions/runs/6606102537/job/17942015613?pr=528

and we want to test OCP on u/s as the operator should support both vanilla k8s and OCP and we want to catch bugs as close to the source as possible

Reply:
I know we want to catch bugs early as possible, however the question I'm posing is: should we really test downstream stuff in the upstream repo? Can't we have something that is doing that separately (or as separate as possible) from upstream?

Reply:
In summary, my view on this is: either we are missing some tests in upstream that don't tie into OCP, or we need to add those OCP-specific tests to downstream if the things it tests for are downstream specific.

Reply:
I am sorry I don't think I am following here.

This is like Openstack that it can run on multiple platforms and every vendor that runs it wants to be sure a PR doesn't break is OpenStack deployment.

so here is the same case we treat OCP as a vendor CI meaning any pull request with bug fixes or new features should work on both vanilla k8s and openshift. there is no point in waiting for the changes to be synced into the openshift repos to find bugs.

that will be the same expectation if now k3s when to be sure the operator works on is platform and continue to work.